plugins/lookup/passwordstore.py

```diff
@@ -175,17 +182,29 @@ def parse_params(self, term):
                 else:
                     raise AnsibleError("{0} is not a correct value for length".format(self.paramvals['length']))
 
+            # Collect pass environment variables from the plugin's parameters.
+            self.env = os.environ.copy()
+
             # Set PASSWORD_STORE_DIR if directory is set
-            if self.paramvals['directory']:
+            if 'directory' in self.paramvals:
```

Comment:
I think that this change is causing the breakage. `directory` is always in `self.paramvals`, see https://github.com/ansible-collections/community.general/blob/141c151f33791888f9ebfda7ff00fce125d5dd26/plugins/lookup/passwordstore.py#L282. You have to do the old check, whether it is falsy or not.

Reply:
spot on, checks are passing now. 
I've also another suggestion: I think the correct way to do the if check would be this:
``if 'directory' in self.paramvals and self.paramvals['directory'] is not None:``

Reply:
That might break backwards compatibility, when someone provided `''` instead of a value. So far, an empty string was equivalent to `None`.

Reply:
All right, it's getting to elaborate now for this small issue, sticking with what works is indeed the best solution for now. Resolving.